Sources/SmokeDynamoDB/AWSDynamoDBCompositePrimaryKeyTable+updateItems.swift

```diff
@@ -139,48 +139,57 @@ public extension AWSDynamoDBCompositePrimaryKeyTable {
         
         return EventLoopFuture.andAllSucceed(futures, on: self.eventLoop)
     }
-    func writeChunkedItemsWithoutThrowing<AttributesType, ItemType>(_ entries: [WriteEntry<AttributesType, ItemType>]) async throws
-    -> [Int: BatchStatementError] {
+    
+    func writeChunkedItemsWithoutThrowing<AttributesType, ItemType>(_ entries: [WriteEntry<AttributesType, ItemType>])
+    -> EventLoopFuture<Set<BatchStatementErrorCodeEnum>> {
         // if there are no items, there is nothing to update
         
         guard entries.count > 0 else {
             self.logger.info("\(entries) with count = 0")
-            return [:]
-        }
-        
-        let statements: [BatchStatementRequest] = try entries.map { try entryToBatchStatementRequest( $0 ) }
-
-        let executeInput = BatchExecuteStatementInput(statements: statements)
-        
-        let result =  try await dynamodb.batchExecuteStatement(input: executeInput)
-
-        guard let responses = result.responses else {
-            self.logger.warning("BatchExecuteStatementOutput: \(result) does not contain responses")
-            return [:]
+            let promise = self.eventLoop.makePromise(of: Set<BatchStatementErrorCodeEnum>.self)
+            promise.succeed(Set())
+            return promise.futureResult
         }
         
-        var failedList: [Int: BatchStatementError] = [:]
-
-        for (index, response) in responses.enumerated() {
-            if let error = response.error {
-                failedList[index] = error
+        do {
+            let statements: [BatchStatementRequest] = try entries.map { try entryToBatchStatementRequest( $0 ) }
+            let executeInput = BatchExecuteStatementInput(statements: statements)
+            return dynamodb.batchExecuteStatement(input: executeInput).map { result -> Set<BatchStatementErrorCodeEnum> in
+                var errorSet: Set<BatchStatementErrorCodeEnum> = Set()
+                result.responses?.forEach { response in
+                    if let error = response.error {
```

Comment:
You can do this in one line since you only care about the error code
`if let error = response.error, let code = error.code {...}`